src/auth/user/user.service.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { User } from './entity/user.model';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import * as bcrypt from 'bcryptjs';
+import { ERROR_MESSAGES, SALT_ROUNDS } from '@app/common/constants/constants';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  async createUser(createUserDto: CreateUserDto): Promise<User> {
+    try {
+      const salt = await bcrypt.genSalt(SALT_ROUNDS);
+      const hashedPassword = await bcrypt.hash(createUserDto.password, salt);
+      createUserDto.password = hashedPassword;
+      const newUser = this.userRepository.create(createUserDto);
+      if (!newUser) {
+        Logger.error('Error creating user');
+        throw new InternalServerErrorException(ERROR_MESSAGES.CREATE_USER);
+      }
+      await this.userRepository.save(newUser);
+      return newUser;
+    } catch (error) {
+      Logger.error('Error creating user', error);
+      throw new InternalServerErrorException(ERROR_MESSAGES.CREATE_USER);
```

Comment:
Remove the logger error. When throwing the error it'll print out the error message regardless in the catch block